### ISSUE TITLE
Runtime error when starting Traefik with a dev plugin

### ISSUE THREAD

**Issue by pipe01** [traefik/traefik#7300]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I'm developing a plugin for pushing HTTP2 files, and I'm testing it using the "devPlugin" section in the static config. After making some changes and starting Traefik to see if they work, I was met with a `runtime error: invalid memory address or nil pointer dereference` coming from Traefik itself when loading the plugin, not my code. The code of the plugin can be found [here](https://github.com/pipe01/plugin-h2push/blob/master/h2push.go).

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.3.0-rc6
Codename:     picodon
Go version:   go1.15.2
Built:        2020-09-16T15:20:51Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Running Traefik with Docker on Windows with the WSL2 backend (issue persists when running the Windows binary).

Static:
```yaml
entryPoints:
    web:
        address: :80

providers:
    file:
        filename: /etc/traefik/traefik.dyn.yaml

log:
  level: DEBUG

pilot:
    token: ****

experimental:
    devPlugin:
        goPath: /gopath
        moduleName: github.com/pipe01/plugin-h2push
```

Dynamic:
```yaml
http:
  services:
    webserver:
      loadBalancer:
        servers:
        - url: 'http://webserver:80'

  routers:
    my-router:
      rule: "Path(`/`)"
      service: webserver
      middlewares:
        - push

  middlewares:
    push:
      plugin:
        dev:
          files:
          - url: url1
            match: match1
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
time="2020-09-17T12:38:58Z" level=info msg="Configuration loaded from file: /etc/traefik/traefik.yaml"
time="2020-09-17T12:38:58Z" level=info msg="Traefik version 2.3.0-rc6 built on 2020-09-16T15:20:51Z"
time="2020-09-17T12:38:58Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"web\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":10000000000},\"respondingTimeouts\":{\"idleTimeout\":180000000000}},\"forwardedHeaders\":{},\"http\":{}}},\"providers\":{\"providersThrottleDuration\":2000000000,\"file\":{\"watch\":true,\"filename\":\"/etc/traefik/traefik.dyn.yaml\"}},\"log\":{\"level\":\"DEBUG\",\"format\":\"common\"},\"pilot\":{\"token\":\"a2ee399b-e932-4ae9-900d-aadf8eb8707c\"},\"experimental\":{\"devPlugin\":{\"goPath\":\"/gopath\",\"moduleName\":\"github.com/pipe01/plugin-h2push\"}}}"
time="2020-09-17T12:38:58Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/contributing/data-collection/\n"
2020/09/17 12:38:58 traefik.go:76: command traefik error: github.com/pipe01/plugin-h2push: failed to import plugin code "github.com/pipe01/plugin-h2push": runtime error: invalid memory address or nil pointer dereference
```

**Comment by pipe01** [comment]
I've found that the error is caused by explicitly assigning `nil` to a `*regexp.Regexp` field.

The line causing the error is https://github.com/pipe01/plugin-h2push/blob/master/h2push.go#L53, specifically assigning `nil` to the `MatchRegexp *regexp.Regexp` field. If the constructor is replaced with `pushFile{H2PushFile: f}` the issue goes away.

**Comment by jspdown** [comment]
I'm able to reproduce the issue on v2.3.0-rc6 but not after v2.3.3. I've also tested it on v2.3.7 and v2.4.0-rc2 and I can no longer reproduce. The issue was fixed when Yaegi has been updated to v0.9.5 (https://github.com/traefik/yaegi/compare/v0.9.4...v0.9.5)
